File: src/main/java/com/university/foreignstudent/entity/Alumni.java
package com.university.foreignstudent.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 校友实体类
 * 用于管理已毕业学生的校友信息
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@TableName("alumni")
public class Alumni {
    
    /**
     * 校友ID，主键，自增
     */
    @TableId(type = IdType.AUTO)
    private Long id;
    
    /**
     * 学生ID（关联学籍表）
     */
    private Long studentId;
    
    /**
     * 姓名
     */
    private String name;
    
    /**
     * 学号
     */
    private String studentNumber;
    
    /**
     * 专业
     */
    private String major;
    
    /**
     * 毕业日期
     */
    private LocalDate graduationDate;
    
    /**
     * 学位：BACHELOR（学士）、MASTER（硕士）、DOCTOR（博士）
     */
    private String degree;
    
    /**
     * 毕业去向：EMPLOYED（就业）、STUDYING（继续深造）、OTHER（其他）
     */
    private String status;
    
    /**
     * 工作单位/继续深造学校
     */
    private String organization;
    
    /**
     * 职位/专业
     */
    private String position;
    
    /**
     * 联系方式（邮箱/电话）
     */
    private String contact;
    
    /**
     * 备注
     */
    private String remarks;
    
    /**
     * 创建时间
     */
    private LocalDateTime createTime;
    
    /**
     * 更新时间
     */
    private LocalDateTime updateTime;
    
    /**
     * 逻辑删除标记：0-未删除，1-已删除
     */
    private Integer deleted;
}

File: frontend/src/api/alumni.ts
import http from './http'
import type { Alumni } from '../types/entity'

/**
 * 获取所有校友记录列表
 */
export function getAllAlumni(): Promise<Alumni[]> {
  return http.get('/alumni')
}

/**
 * 根据ID获取校友记录
 */
export function getAlumniById(id: number): Promise<Alumni> {
  return http.get(`/alumni/${id}`)
}

/**
 * 根据学生ID获取校友记录
 */
export function getAlumniByStudentId(studentId: number): Promise<Alumni> {
  return http.get(`/alumni/student/${studentId}`)
}

/**
 * 创建校友记录
 */
export function createAlumni(alumni: Partial<Alumni>): Promise<Alumni> {
  return http.post('/alumni', alumni)
}

/**
 * 从学籍记录同步创建校友记录（毕业同步）
 */
export function syncAlumniFromStudentRecord(data: {
  studentId: number
  graduationDate: string
  degree?: string
  status?: string
  organization?: string
  position?: string
  contact?: string
}): Promise<Alumni> {
  return http.post('/alumni/sync', data)
}

/**
 * 更新校友记录
 */
export function updateAlumni(id: number, alumni: Partial<Alumni>): Promise<Alumni> {
  return http.put(`/alumni/${id}`, alumni)
}

/**
 * 删除校友记录
 */
export function deleteAlumni(id: number): Promise<void> {
  return http.delete(`/alumni/${id}`)
}

